### ISSUE TITLE
Maccy notifies me every time I copy something

### ISSUE THREAD

**Issue by vegerot** [p0deje/Maccy#692]
### Before Submitting Your Bug Report

- [X] I have verified that there isn't already an issue reporting the same bug to prevent duplication.
- [X] I have seen the [FAQ](https://github.com/p0deje/Maccy?tab=readme-ov-file#faq).

### Maccy Version (see 'About' window)

How do I see the about window?

### macOS Version

13.6.1

### Maccy Settings

```Shell
{
    "KeyboardShortcuts_delete" = "{\\"carbonModifiers\\":2048,\\"carbonKeyCode\\":51}";
    "KeyboardShortcuts_pin" = "{\\"carbonModifiers\\":2048,\\"carbonKeyCode\\":35}";
    "KeyboardShortcuts_popup" = "{\\"carbonModifiers\\":768,\\"carbonKeyCode\\":34}";
    "LaunchAtLogin__hasMigrated" = 1;
    "NSWindow Frame SUStatusFrame" = "1480 1298 400 134 0 0 3360 1865 ";
    "NSWindow Frame SUUpdateAlert" = "1370 1100 620 398 0 0 3360 1865 ";
    "NSWindow Frame com.sindresorhus.Settings.FrameAutosaveName" = "1409 1112 542 399 0 0 3360 1865 ";
    SUEnableAutomaticChecks = 1;
    SUHasLaunchedBefore = 1;
    SULastCheckTime = "2023-12-06 22:32:00 +0000";
    SUUpdateGroupIdentifier = 1475230968;
    WebKitDefaultFontSize = 13;
    WebKitJavaScriptEnabled = 0;
    WebKitPluginsEnabled = 0;
    WebKitStandardFont = "-apple-system-font";
    WebKitUserStyleSheetEnabledPreferenceKey = 1;
    WebKitUserStyleSheetLocationPreferenceKey = "/Applications/Maccy.app/Contents/Frameworks/Sparkle.framework/Resources/ReleaseNotesColorStyle.css";
    enabledPasteboardTypes =     (
        "public.tiff",
        "public.rtf",
        "public.file-url",
        "public.utf8-plain-text",
        "public.png",
        "public.html"
    );
    historySize = 999;
    ignoredPasteboardTypes =     (
        "com.agilebits.onepassword",
        "com.typeit4me.clipping",
        "Pasteboard generator type",
        "de.petermaurer.TransientPasteboardType",
        "net.antelle.keeweb"
    );
    menuIcon = clipboard;
    migrations =     {
        "2020-04-25-allow-custom-ignored-types" = 1;
        "2020-06-19-use-keyboardshortcuts" = 1;
        "2020-09-01-ignore-keeweb" = 1;
        "2021-02-20-allow-to-customize-supported-types" = 1;
        "2021-06-28-add-title-to-history-item" = 1;
        "2021-10-16-remove-dynamic-pasteboard-types" = 1;
        "2022-08-01-rename-suppress-clear-alert" = 1;
        "2022-11-14-add-html-rtf-to-supported-types" = 1;
        "2023-01-22-add-regexp-search-mode" = 1;
    };
    pasteByDefault = 1;
    searchMode = fuzzy;
    showRecentCopyInMenuBar = 1;
}
```

### Description

For some reason, starting this morning every time I copy something Maccy sends a notification telling me what I just copied

### Steps to Reproduce

1. Copy a thing

Expected: Should not get a notification

Actual: received notification

<img width="387" alt="image" src="https://github.com/p0deje/Maccy/assets/5297725/d1c11601-8bf9-41c0-be5a-aefb654310c9">

**Comment by p0deje** [comment]
You can disable notifications for Maccy in System Settings / Notifications.

**Comment by vegerot** [comment]
Thanks.  This seems like a weird default though, right?  Why would someone want to be notified whenever they copy something?

**Comment by p0deje** [comment]
When Maccy starts it asks whether you want to allow notifications or not. If you agree to allow, it would start sending them. If you don't allow or simply close the allow notification - it wouldn't. I guess it's not an ideal behavior, but sort of a default one for most apps.

**Comment by domoritz** [comment]
I find this behavior confusing as well. Allowing notifications to me just means that an app should be allowed to send any notifications. However, I would expect that I can disable specific kinds of notifications in the app.

**Comment by tgdn** [comment]
I also find this confusing, and the noise can become quite annoying without an easy way to disable it.

**Comment by p0deje** [comment]
Would you it be okay to have a button in Maccy Preferences that takes you to System Settings / Notifications / Maccy where you can configure it all?

**Comment by cotwitch** [comment]
Dear Alex,
in my opinion, there should be a preference option to choose if copied content should be "sent to notifications" or not. 

What if you (as a developer) will need (in a future) to notify users about something really important (like a new version of the application or similar) and lot (maybe most?) of users will have notifications disabled? 

Notifications settings in macOS are "global" (so, it disables ALL the Maccy notifications). 

But me, as an happy user 🥳 of Maccy - I don't want to disable ALL the Maccy notifications - as I'm still curious 👀 about any other, meaningful notifications from Maccy...
I just want to disable (solely) showing annoying notification when anything is copied into the clipboard.

Kindly thank you for consideration 🎩

**Comment by vegerot** [comment]
silly question, but why is this even a feature?  Why would one of your users want to be notified every time they copy something?

**Comment by p0deje** [comment]
Asked in #211

**Comment by p0deje** [comment]
To be frank, the current implementation doesn't fully utilize a notification center. For example, sounds are played as plain sounds because I've received multiple bug reports saying that custom sounds are not used and system beep is used instead. Hence, sounds don't respect DND for example. This needs to be fixed.

I agree it would be best to implement a configuration for each notification type, yet I don't know how useful it would be compared to the effort of developing it (and notably translating to all the different languages Maccy supports). IMO it's too much work for little value.

I will add a button "Configure Notifications" in Maccy preferences that takes you to System Settings for now.  I will also consider implementing more advanced configurations for each notification type if Maccy adds more notifications at some point.

**Comment by domoritz** [comment]
What was the original motivation for having notifications for copied items? Maybe it should be  off by default until someone presses the "configure notifications" button?

**Comment by cotwitch** [comment]
By the way, for example, Fluor application, which I'm also using uses following configuration/preferences: 

<img width="551" alt="Screenshot 2023-12-29 at 01 55 58" src="https://github.com/p0deje/Maccy/assets/10708898/d5f87afe-936d-4e84-ae51-c377661249f5">

**Comment by p0deje** [comment]
In the upcoming version, there will be a link to notifications/sounds configuration in the bottom of General preferences:

<img width="654" alt="Screenshot 2023-12-28 at 19 01 37" src="https://github.com/p0deje/Maccy/assets/665846/c15c7fa3-9bfe-4c02-b333-500ecad1938f">

I'm going to close this for now, see https://github.com/p0deje/Maccy/issues/692#issuecomment-1871424363 for an explanation. It doesn't seem to be possible to disable sounds in DND, let's see if Apple improves custom sounds in notifications in the future so we could just use it instead.

**Comment by findepi** [comment]
> When Maccy starts it asks whether you want to allow notifications or not. If you agree to allow, it would start sending them.

just sharing my user perspective. Some future Maccy version might have a notification e.g for "new release with bug fixes available". I may want to get it, but I won't if i disable all notifications in the notifications center. As an engaged user would prefer fine-grained controls within the app. I disable notifications in Mac's notifications center only if an app annoys me :) 



anyway, thanks for Maccy and maintaining it! it's awesome!

**Comment by domoritz** [comment]
That's well put and describes why I originally came here to report a bug. Disabling notifications is a way for me to defend against applications that notify too much. It feels like a last resort rather than a setting.

**Comment by marcelarie** [comment]
The notifications should be enabled by default to allow the app to notify users about important things like new updates or errors, but not on every copy.

**Comment by jimmyhmiller** [comment]
I just spent an hour trying to debug where this sound was coming from. I had turned maacy notifications to None, rather than turning them off. I could not correlate as I as working that it was copying things that was making this really annoying noise. 

I maybe would have found this if the sound setting had been in maacy itself, rather than a side-effect of notifications being enabled. It also seems weird that the sound would play when it is set to none.

Do people actually want these noises?

**Comment by p0deje** [comment]
>  It also seems weird that the sound would play when it is so to none.

I think that's allowed - you can hide visual alerts but keep sounds.


> I maybe would have found this if the sound setting had been in maacy itself, rather than a side-effect of notifications being enabled.

The link in Maccy general preferences is called "Notifications & sounds". 

<img width="548" alt="Screenshot 2024-03-18 at 12 22 01" src="https://github.com/p0deje/Maccy/assets/665846/3d090247-6595-4350-ab7d-5bb2f75e81b8">

*sigh* I guess I need to implement a separate settings pane, but it's too much at the moment for me.

**Comment by jimmyhmiller** [comment]
@p0deje I totally get it from your perspective. It is possible to change these settings. If this were some chat app I would have been able to figure it out. But I copy and paste a lot. It is just like typing to me. I just never would have guessed that a clipboard manager would play sounds as "notifications" in response to my actions. Notifications are something I think of as happening separate from my own actions, a result of something happening in the background, or someone sending me a message.

**Comment by jimmyhmiller** [comment]
@p0deje No worries, I get that people want different things. Thanks for the great piece of software.

**Comment by kylecordes** [comment]
Stumbling on this old item when searching for the same question myself a couple years later:

* I also found it surprising that the default is to notify on every copy.
* I also found it surprising that the adjustment is to _block the application's ability to send notifications entirely_, rather than asking the application not to notify of this specific event.

It works great, but this bit is potentially worthy of enhancement!
